addons/event/models/mail_template.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+from odoo.osv import expression
+
+
+class MailTemplate(models.Model):
+    _inherit = 'mail.template'
+
+    @api.model
+    def _name_search(self, name, args=None, operator='ilike', limit=100, name_get_uid=None):
+        """Ugly trick to add a domain on a reference field.
```

Comment:
I would remove Ugly trick, as we should always be proud of what we achieve ! Joke apart, this is not so ugly. Maybe rephase to something like "Context-based hack to filter reference field in a m2o search box to emulate a domain the ORM currently does notn support" .